### ISSUE TITLE
Difference between the edit() and edit_dataset()

### ISSUE THREAD

**Issue by Wang-Z-W** [zjunlp/EasyEdit#595]
When I try to run IKE on MMEdit, I have the following question.

Why are there two different edit functions in MultimodalEditor?

When calculating the locality metric, in edit(), it's
```
base_logits = torch.tensor(metrics['pre']['locality_output']).to(torch.float32)
post_logits = torch.tensor(metrics['post']['locality_output']).to(torch.float32)
metrics['post']['locality_acc'] = sum(post_logits.view(-1) == base_logits.view(-1))/base_logits.view(-1).shape[0]
```
while in edit_dataset(), it's
```
base_logits = metrics['pre']['locality_output'].to(torch.float32)
post_logits = metrics['post']['locality_output'].to(torch.float32)
if post_logits.shape[1] > base_logits.shape[1]:
        post_logits = post_logits[:, -base_logits.shape[1]:, :]
else:
        base_logits = base_logits[:, -post_logits.shape[1]:, :]

base_logits_softmax_top_k = torch.topk(torch.nn.functional.softmax(base_logits, dim=-1), k=1, dim=-1).indices
post_base_logits_softmax_top_k = torch.topk(torch.nn.functional.softmax(post_logits, dim=-1), k=1, dim=-1).indices
metrics['post']['locality_acc'] = sum(post_base_logits_softmax_top_k.view(-1) == base_logits_softmax_top_k.view(-1))/post_base_logits_softmax_top_k.view(-1).shape[0]
```

Obviously, there are extra judgment and truncation operations. I have no idea why this difference exists.

**Comment by zxlzr** [comment]
do you have any further questions?

**Comment by Wang-Z-W** [comment]
Thank you so much! I have no further question.
